dbfit-java/core/src/main/java/dbfit/api/AbstractDbEnvironment.java

```diff
@@ -117,23 +116,41 @@ public final PreparedStatement createStatementWithBoundFixtureSymbols(
                 cs.setObject(i + 1, value);
             }
         }
-        return cs;
+
+        return createPreparedStatement(cs);
     }
 
     @Override
-    public DdlStatementExecution createDdlStatementExecution(String ddl)
+    public DdlStatement createDdlStatement(String ddl)
             throws SQLException {
-        return new DdlStatementExecution(getConnection().createStatement(), ddl);
+        return new DdlStatement(getConnection().createStatement(), ddl);
+    }
+
+    /**
+     * Create a {@link DbStatement} for the given prepared statement.
+     * This is the method to override if custom implementation needs to be returned
+     */
+    protected DbStatement createPreparedStatement(PreparedStatement statement) {
+        return new DbStatement(statement);
     }
 
     @Override
-    public StatementExecution createStatementExecution(PreparedStatement statement) {
-        return new StatementExecution(statement);
+    public final DbStatement createPreparedStatement(String commandText) throws SQLException {
+        return createPreparedStatement(getConnection().prepareStatement(commandText));
+    }
+
+    /**
+     * Create a {@link DbStatement} for the given prepared statement.
+     * This is the method to override if custom implementation needs to be returned
+     */
+    protected DbStatement createCallStatement(PreparedStatement statement, boolean isFunction)
+            throws SQLException {
+        return new DbStatement(statement);
     }
 
     @Override
-    public StatementExecution createFunctionStatementExecution(PreparedStatement statement) {
-        return new StatementExecution(statement);
+    public final DbStatement createCallStatement(String commandText, boolean isFunction) throws SQLException {
```

Comment:
> ...and then createCallDbStatement (or createCallableDbStatement)?

I had some hesitation about extra particles (`Execution`, `Db`) since they're making names longer. Maybe `Db` is not so huge trouble.